### PR TITLE
Allow azure signers to _not_ perform hashing on data to sign

### PR DESCRIPTION
**Pull request by rain-on**

### CODE REVIEW

File: signing/secp256k1/impl/src/test/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerTest.java [hunk 1]
```diff
@@ -41,7 +55,35 @@ public void azureSignerCanSignTwice() {
...
-    signer.sign("Hello World".getBytes(UTF_8));
-    signer.sign("Hello World".getBytes(UTF_8));
+
+    final byte[] dataToHash = "Hello World".getBytes(UTF_8);
+    signer.sign(dataToHash);
+    signer.sign(dataToHash);
+  }
+
+  @Test
+  void azureWithOutHashingProducesDifferentAnswerToThatWithHashing() throws SignatureException {
```
Comment (jframe) [inline signing/secp256k1/impl/src/test/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerTest.java:65]: Can this test name be improved? Isn't this testing that the AzureKeyVaultSigner isn't hashing the data? azureWithoutHashingDoesntHashData?

File: signing/secp256k1/impl/src/main/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerFactory.java [hunk 2]
```diff
@@ -29,12 +29,18 @@
...
-
   private static final Logger LOG = LogManager.getLogger();
 
+  private final boolean needsToHash;
+
+  public AzureKeyVaultSignerFactory() {
+    this(true);
+  }
+
+  public AzureKeyVaultSignerFactory(boolean needsToHash) {
```
Comment (jframe) [inline signing/secp256k1/impl/src/main/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerFactory.java:40]: nit: final

File: signing/secp256k1/impl/src/test/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerTest.java [hunk 3]
```diff
@@ -27,6 +37,10 @@
   private static final String keyVaultName = System.getenv("AZURE_KEY_VAULT_NAME");
   private static final String tenantId = System.getenv("AZURE_TENANT_ID");
 
+  public static final String PUBLIC_KEY_HEX_STRING =
+      "09b02f8a5fddd222ade4ea4528faefc399623af3f736be3c44f03e2df22fb792f3931a4d9573d333ca74343305762a753388c3422a86d98b713fc91c1ea04842";
+  final Bytes publicKeyBytes = Bytes.fromHexString(PUBLIC_KEY_HEX_STRING);
```
Comment (jframe) [inline signing/secp256k1/impl/src/test/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerTest.java:42]: nit: private

File: signing/secp256k1/impl/src/test/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerTest.java (hunk 1 above)
Comment (rain-on) [reply signing/secp256k1/impl/src/test/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerTest.java:65]: true true.

File: signing/secp256k1/impl/src/main/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerFactory.java (hunk 2 above)
Comment (rain-on) [reply signing/secp256k1/impl/src/main/java/tech/pegasys/signers/secp256k1/azure/AzureKeyVaultSignerFactory.java:40]: done

**Pull request merged**